libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo3/ast/gqltype.kt

```diff
@@ -81,3 +83,45 @@ internal fun GQLType.isOutputType(typeDefinitions: Map<String, GQLTypeDefinition
         || it is GQLEnumTypeDefinition
   }
 }
+
+private fun GQLNullability?.selfNullability(): GQLNullability? {
+  return when (this) {
+    is GQLListNullability -> this.selfNullability
+    else -> this
+  }
+}
+
+private fun GQLType.withListCcn(nullability: GQLNullability?): GQLType {
+  if (this is GQLListType && nullability is GQLListNullability) {
+    return copy(type = type.withCcn(nullability.itemNullability))
+  } else if (this is GQLListType && nullability !is GQLListNullability) {
+    return this
+  } else if (this !is GQLListType && nullability is GQLListNullability) {
+    return this
+  } else if (this !is GQLListType && nullability !is GQLListNullability) {
+    return this
+  } else {
+    error("")
+  }
+}
+
+@ApolloExperimental
+fun GQLType.withCcn(nullability: GQLNullability?): GQLType {
```

Comment:
Nitpicking again, but we don't use "ccn" before so maybe `withNullability` / `withListNullability` would be slightly more consistent?

Reply:
PR updated. Although at the moment I really like the idea of:

```kotlin
fun GQLType.applyModifier(modifier: GQLTypeModifier?): GQLType 
```

But maybe I've just been doing too much Jetpack Compose 😄

Reply:
Although `apply` kind of hints that the instance is modified in-place so maybe more 

```kotlin
fun GQLType.withModifier(modifier: GQLTypeModifier?): GQLType 
```